src/features/unified-share-modal/ContactRestrictionNotice.js

```diff
@@ -54,12 +70,15 @@ const ContactRestrictionNotice = ({
     const selectedValue = getProp(selectedJustificationReason, 'value', null);
     const isErrorTooltipShown = !!error;
 
-    const restrictionNoticeMessage = isRestrictionJustificationEnabled
-        ? messages.justifiableContactRestrictionNotice
-        : messages.contactRestrictionNotice;
+    const justificationStatus = isRestrictionJustificationEnabled
+        ? RESTRICTION_JUSTIFICATION_ENABLED
+        : RESTRICTION_JUSTIFICATION_DISABLED;
+    const restrictedContactCountType = restrictedExternalContactCount === 1 ? SINGLE_CONTACT : MULTIPLE_CONTACTS;
+
     const removeButtonLabelMessage = isRestrictionJustificationEnabled
         ? messages.justifiableContactRestrictionRemoveButtonLabel
         : messages.contactRestrictionRemoveButtonLabel;
+    const restrictionNoticeMessage = restrictionNoticeMessageMap[justificationStatus][restrictedContactCountType];
 
     const justificationSelectSection = isFetchingJustificationReasons ? (
         <LoadingIndicator className="bdl-ContactRestrictionNotice-loadingIndicator" />
```

Comment:
I also took the opportunity to remove `ContactsEmailsTooltip` which will no longer be used. Interpolating React components within translatable messages requires React intl 3+ and no suitable workaround was found.